cmd/system-probe/subcommands/run/command.go

```diff
@@ -81,6 +82,7 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 				fx.Supply(sysprobeconfigimpl.NewParams(sysprobeconfigimpl.WithSysProbeConfFilePath(globalParams.ConfFilePath))),
 				fx.Supply(logimpl.ForDaemon("SYS-PROBE", "log_file", common.DefaultLogFile)),
 				fx.Supply(rcclient.Params{AgentName: "system-probe", AgentVersion: version.AgentVersion}),
+				fx.Supply(optional.NewNoneOption[secrets.Component]()),
```

Comment:
Does this mean components which do not need an optional component must specify they do not need it?

Reply:
Yes, correct, the `None` value must be explicitly provided. This is because fx is not aware of the semantics of `optional`, so it requires a constructor for this type.